### PR TITLE
Fix Android spawn & Upgrade Frida v12.2.6 & Breakpoints help added & …

### PR DESCRIPTION
**Pull request by enovella**
- Fix Android spawn
- Upgrade Frida v12.2.6
- Breakpoints help added & Order commands

### CODE REVIEW

File: src/io_frida.c [hunk 1]
```diff
@@ -160,7 +160,7 @@ static RIODesc *__open(RIO *io, const char *pathname, int rw, int mode) {
 		}
 
 		FridaSpawnOptions *options = frida_spawn_options_new ();
-		frida_spawn_options_set_argv (options, argv, g_strv_length (argv));
+		//frida_spawn_options_set_argv (options, argv, g_strv_length (argv));
```
[the remark this answers is not in the document]
Comment (trufae) [reply src/io_frida.c]: oh nvm, this will affect host depending on target.

[the remark this answers is not in the document]
Comment (enovella) [reply src/io_frida.c]: This was exactly was I was discussing with you but it didn't solve the problem. Though, I can give another try along today.

Cheers

**Comment by enovella** [comment]
Hi @trufae,

perhaps we can follow the Frida code style and do something cleaner such as:
```c
if (options.has_argv) {
    frida_spawn_options_set_argv (options, argv, g_strv_length (argv));
}

File: src/io_frida.c [hunk 2]
```diff
@@ -160,7 +160,9 @@ static RIODesc *__open(RIO *io, const char *pathname, int rw, int mode) {
 		}
 
 		FridaSpawnOptions *options = frida_spawn_options_new ();
-		frida_spawn_options_set_argv (options, argv, g_strv_length (argv));
+		if (argv[1]) {
```
Comment (radare) [inline src/io_frida.c]: do it like this:
```c
$ git diff
diff --git a/src/io_frida.c b/src/io_frida.c
index b7d653f..c7f4e3d 100644
--- a/src/io_frida.c
+++ b/src/io_frida.c
@@ -160,7 +160,10 @@ static RIODesc *__open(RIO *io, const char *pathname, int rw, int mode) {
                }

                FridaSpawnOptions *options = frida_spawn_options_new ();
-               frida_spawn_options_set_argv (options, argv, g_strv_length (argv));
+               const int argc = g_strv_length (argv);
+               if (argc > 1) {
+                       frida_spawn_options_set_argv (options, argv, g_strv_length (argv));
+               }
                rf->pid = frida_device_spawn_sync (rf->device, argv[0], options, &error);
                g_object_unref (options);
                r_str_argv_free (argv);
$
```

Comment (enovella) [reply src/io_frida.c]: Maybe shorter ?
```c
const int argc = g_strv_length (argv);
if (argc > 1) {
	frida_spawn_options_set_argv (options, argv, argc);
}
```

Comment (enovella) [reply src/io_frida.c]: There you go!

**Pull request merged**